samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -267,20 +273,41 @@ public TestRunner addOutputStream(CollectionStream stream) {
     return this;
   }
 
+
   /**
    * Utility to run a test configured using TestRunner
+   *
+   * @param timeout in ms for the Test to Run
+   * @throws InterruptedException if the current thread was interrupted
+   * @throws ExecutionException if the samza job threw an exception
+   * @throws TimeoutException if the wait timed out while running Samza job
    */
-  public void run() {
+  public void run(int timeout) throws InterruptedException, ExecutionException, TimeoutException {
```

Comment:
Integer timeout doesn't clearly define the unit of granularity(`seconds` or `milliseconds` or `minutes`) of the parameter and how to interpret it for the user.
Would be better to take `TimeUnit` as a parameter here along with the int timeout or simply `Duration` as a parameter.
Please change if it makes sense.

Reply:
Will make this change!